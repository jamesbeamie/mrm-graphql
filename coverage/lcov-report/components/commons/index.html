<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/commons</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/commons
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>48/271</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">7.59% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>11/145</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/69</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.68% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>48/257</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="ActionButtons.js"><a href="ActionButtons.js.html">ActionButtons.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="BackArrow.js"><a href="BackArrow.js.html">BackArrow.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	</tr>

<tr>
	<td class="file high" data-value="Button.js"><a href="Button.js.html">Button.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Calendar.js"><a href="Calendar.js.html">Calendar.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	</tr>

<tr>
	<td class="file low" data-value="CheckboxSlide.js"><a href="CheckboxSlide.js.html">CheckboxSlide.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="32" class="abs low">8/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="26.67" class="pct low">26.67%</td>
	<td data-value="30" class="abs low">8/30</td>
	</tr>

<tr>
	<td class="file medium" data-value="DataNotFound.js"><a href="DataNotFound.js.html">DataNotFound.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file low" data-value="Dropdown.js"><a href="Dropdown.js.html">Dropdown.js</a></td>
	<td data-value="12.5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	</tr>

<tr>
	<td class="file low" data-value="EpicTowerInputs.js"><a href="EpicTowerInputs.js.html">EpicTowerInputs.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="ErrorBoundary.js"><a href="ErrorBoundary.js.html">ErrorBoundary.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	</tr>

<tr>
	<td class="file high" data-value="ErrorIcon.js"><a href="ErrorIcon.js.html">ErrorIcon.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file low" data-value="ExportButton.js"><a href="ExportButton.js.html">ExportButton.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	</tr>

<tr>
	<td class="file low" data-value="IconButtons.js"><a href="IconButtons.js.html">IconButtons.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	</tr>

<tr>
	<td class="file low" data-value="Input.js"><a href="Input.js.html">Input.js</a></td>
	<td data-value="11.76" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	</tr>

<tr>
	<td class="file low" data-value="MenuTitle.js"><a href="MenuTitle.js.html">MenuTitle.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Modal.js"><a href="Modal.js.html">Modal.js</a></td>
	<td data-value="7.14" class="pic low"><div class="chart"><div class="cover-fill" style="width: 7%;"></div><div class="cover-empty" style="width:93%;"></div></div></td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="MrmModal.js"><a href="MrmModal.js.html">MrmModal.js</a></td>
	<td data-value="57.14" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 57%;"></div><div class="cover-empty" style="width:43%;"></div></div></td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="Overlay.js"><a href="Overlay.js.html">Overlay.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="Pagination.js"><a href="Pagination.js.html">Pagination.js</a></td>
	<td data-value="9.68" class="pic low"><div class="chart"><div class="cover-fill" style="width: 9%;"></div><div class="cover-empty" style="width:91%;"></div></div></td>
	<td data-value="9.68" class="pct low">9.68%</td>
	<td data-value="31" class="abs low">3/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="30" class="abs low">3/30</td>
	</tr>

<tr>
	<td class="file low" data-value="Placeholder.js"><a href="Placeholder.js.html">Placeholder.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	</tr>

<tr>
	<td class="file low" data-value="SelectImage.js"><a href="SelectImage.js.html">SelectImage.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file medium" data-value="SelectInput.js"><a href="SelectInput.js.html">SelectInput.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Sort.js"><a href="Sort.js.html">Sort.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="12" class="abs low">2/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="12" class="abs low">2/12</td>
	</tr>

<tr>
	<td class="file high" data-value="Spinner.js"><a href="Spinner.js.html">Spinner.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file low" data-value="TheCrestInputs.js"><a href="TheCrestInputs.js.html">TheCrestInputs.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file low" data-value="Tooltip.js"><a href="Tooltip.js.html">Tooltip.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 11 2019 19:58:40 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
