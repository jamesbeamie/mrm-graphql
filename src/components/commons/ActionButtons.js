import React from 'react';
import PropTypes from 'prop-types';
import Spinner from './Spinner';
import Button from '../commons/Button';

import '../../assets/styles/actionbuttons.scss';

/**
 * Reusable button component for performing
 * actions and canceling
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
const ActionButtons = props => (
  props.isLoading
    ? <Spinner size="small" /> :
    <div className="form-action-buttons">
      <Button handleClick={props.onClickCancel} title={props.cancelButtonText} />
      { props.showActionButton &&
      <Button type={2} handleClick={props.onClickSubmit} title={props.actionButtonText} classProp="main-button" />
      }
    </div>
);

ActionButtons.propTypes = {
  onClickCancel: PropTypes.func.isRequired,
  cancelButtonText: PropTypes.string,
  actionButtonText: PropTypes.string,
  isLoading: PropTypes.bool,
  onClickSubmit: PropTypes.func,
  showActionButton: PropTypes.bool,
};

ActionButtons.defaultProps = {
  onClickSubmit: PropTypes.func,
  cancelButtonText: 'CANCEL',
  actionButtonText: 'SAVE CHANGES',
  isLoading: false,
  showActionButton: true,
};

export default ActionButtons;
