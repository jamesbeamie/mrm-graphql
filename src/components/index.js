import NavBar from '../containers/NavBar';
import LoginPage from './login/Login';

export { NavBar, LoginPage };
