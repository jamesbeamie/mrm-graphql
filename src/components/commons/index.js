import SelectInput from './SelectInput';
import Modal from './Modal';
import { Input } from './Input';
import ActionButtons from './ActionButtons';
import IconButtons from './IconButtons';

export { SelectInput, Modal, Input, ActionButtons, IconButtons };
