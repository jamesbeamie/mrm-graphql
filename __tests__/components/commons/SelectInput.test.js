import React from 'react';
import { shallow } from 'enzyme';
import { SelectInput } from '../../../src/components/commons';

describe('SelectInput component', () => {
  const sampleOptions = [
    { id: 1, name: 1, displayText: 'hello' },
    { id: 2, name: 2, displayText: 'how are you' },
  ];
  const handleOptionsSelectChange = jest.fn();
  const handleChildrenSelectChange = jest.fn();
  const optionsWrapper = shallow(<SelectInput
    name="testSelect"
    id="testSelect"
    value="1"
    onChange={handleOptionsSelectChange}
    labelText="Select Something"
    options={sampleOptions}
  />);

  const childrenWrapperCode = (
    <SelectInput
      name="testSelect"
      id="testSelect"
      value="1"
      onChange={handleChildrenSelectChange}
      labelText="Select Something"
    >
      <option value={sampleOptions[0].value}>
        {sampleOptions[0].displayText}
      </option>
      <option value={sampleOptions[1].value}>
        {sampleOptions[1].displayText}
      </option>
    </SelectInput>
  );

  const childrenWrapper = shallow(childrenWrapperCode);

  const emptyWrapper = shallow(<SelectInput
    name="testSelect"
    id="testSelect"
    value="1"
    onChange={handleOptionsSelectChange}
    labelText="Select Something"
  />);

  it('should render properly when given options prop', () => {
    expect(childrenWrapper).toMatchSnapshot();
  });

  it('should render properly when given children', () => {
    expect(optionsWrapper).toMatchSnapshot();
  });

  it('should call console.warn when given neither option nor children', () => {
    global.console.warn = jest.fn();
    expect(emptyWrapper).toMatchSnapshot();
  });

  it('should not render placeholder when its not provided', () => {
    expect(emptyWrapper.find('option')).toHaveLength(0);
  });


  it('should render placeholder when provided', () => {
    emptyWrapper.setProps({ placeholder: 'Hello' });
    emptyWrapper.update();
    const option = emptyWrapper.find('option');
    expect(emptyWrapper).toMatchSnapshot();
    expect(option).toHaveLength(1);
    expect(option.prop('children')[0]).toBe('Hello');
  });

  it('should display value if displayText isn\'t given in options', () => {
    const newOptions = [{ id: 4 }, { id: 5 }];
    optionsWrapper.setProps({ options: newOptions });
    optionsWrapper.update();
    const options = optionsWrapper.find('options');
    options.map((option, index) => (
      expect(option.props().children).toEqual(newOptions[index].value)
    ));
    expect(optionsWrapper).toMatchSnapshot();
  });

  it('should use \'calendarId\' incase \'id\' is not given in option', () => {
    const newOptions = [{ calendarId: 4, name: 'Nairobi' }, { calendarId: 5, name: 'Kampala' }];
    const newOptionsWrapper = shallow(<SelectInput
      name="testSelect"
      id="testSelect"
      value="Nairobi"
      onChange={handleOptionsSelectChange}
      labelText="Select Something"
      options={newOptions}
    />);
    newOptionsWrapper.setProps({ options: newOptions });
    newOptionsWrapper.update();
    const options = newOptionsWrapper.find('options');
    options.map((option, index) => (
      expect(option.props().children).toEqual(newOptions[index].value)
    ));
    expect(newOptionsWrapper).toMatchSnapshot();
  });
});
