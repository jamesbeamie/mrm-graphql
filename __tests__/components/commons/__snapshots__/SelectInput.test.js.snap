// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput component should call console.warn when given neither option nor children 1`] = `
<div
  className="input1"
>
  <label
    htmlFor="testSelect"
  >
    Select Something
    <form
      id="LocationFilters"
    >
      <select
        className="default-select"
        defaultValue="1"
        id="testSelect"
        name="testSelect"
        onChange={[Function]}
        required={false}
        selected={-1}
      >
        You need to provide atleast children or options to the select element
      </select>
    </form>
  </label>
</div>
`;

exports[`SelectInput component should display value if displayText isn't given in options 1`] = `
<div
  className="input1"
>
  <label
    htmlFor="testSelect"
  >
    Select Something
    <form
      id="LocationFilters"
    >
      <select
        className="default-select"
        defaultValue="1"
        id="testSelect"
        name="testSelect"
        onChange={[Function]}
        required={false}
        selected={-1}
      >
        <option
          key="4"
          value={4}
        >
          <Component />
        </option>
        <option
          key="5"
          value={5}
        >
          <Component />
        </option>
      </select>
    </form>
  </label>
</div>
`;

exports[`SelectInput component should render placeholder when provided 1`] = `
<div
  className="input1"
>
  <label
    htmlFor="testSelect"
  >
    Select Something
    <form
      id="LocationFilters"
    >
      <select
        className="default-select"
        defaultValue="1"
        id="testSelect"
        name="testSelect"
        onChange={[Function]}
        required={false}
        selected={-1}
      >
        <option
          value=""
        >
          Hello
           
        </option>
        You need to provide atleast children or options to the select element
      </select>
    </form>
  </label>
</div>
`;

exports[`SelectInput component should render properly when given children 1`] = `
<div
  className="input1"
>
  <label
    htmlFor="testSelect"
  >
    Select Something
    <form
      id="LocationFilters"
    >
      <select
        className="default-select"
        defaultValue="1"
        id="testSelect"
        name="testSelect"
        onChange={[Function]}
        required={false}
        selected={-1}
      >
        <option
          key="1"
          value={1}
        >
          1
        </option>
        <option
          key="2"
          value={2}
        >
          2
        </option>
      </select>
    </form>
  </label>
</div>
`;

exports[`SelectInput component should render properly when given options prop 1`] = `
<div
  className="input1"
>
  <label
    htmlFor="testSelect"
  >
    Select Something
    <form
      id="LocationFilters"
    >
      <select
        className="default-select"
        defaultValue="1"
        id="testSelect"
        name="testSelect"
        onChange={[Function]}
        required={false}
        selected={-1}
      >
        <option>
          hello
        </option>
        <option>
          how are you
        </option>
      </select>
    </form>
  </label>
</div>
`;

exports[`SelectInput component should use 'calendarId' incase 'id' is not given in option 1`] = `
<div
  className="input1"
>
  <label
    htmlFor="testSelect"
  >
    Select Something
    <form
      id="LocationFilters"
    >
      <select
        className="default-select"
        defaultValue="Nairobi"
        id="testSelect"
        name="testSelect"
        onChange={[Function]}
        required={false}
        selected={-1}
      >
        <option
          key="Nairobi"
        >
          Nairobi
        </option>
        <option
          key="Kampala"
        >
          Kampala
        </option>
      </select>
    </form>
  </label>
</div>
`;
